File: problems/1262_greatest_sum_divisible_by_three/solution.py
class Solution:
    def maxSumDivThree(self, nums: list[int]) -> int:
        dp = [0, float('-inf'), float('-inf')]
        for x in nums:
            temp = dp[:]
            for i in range(3):
                dp[(i + x) % 3] = max(dp[(i + x) % 3], temp[i] + x)
        return dp[0]

File: problems/1262_greatest_sum_divisible_by_three/Solution.java
class Solution {
    public int maxSumDivThree(int[] nums) {
        int[] dp = {0, Integer.MIN_VALUE, Integer.MIN_VALUE};
        for (int x : nums) {
            int[] temp = dp.clone();
            for (int i = 0; i < 3; i++)
                dp[(i + x) % 3] = Math.max(dp[(i + x) % 3], temp[i] + x);
        }
        return dp[0];
    }
}
